### PR TITLE
adding leader and marathon leader run capabilities

### PR DESCRIPTION
**Pull request by kensipe**

### CODE REVIEW

**Comment by kensipe** [comment]
Need to API docs and test... manual verification complete.

File: shakedown/dcos/__init__.py [hunk 1]
```diff
@@ -111,6 +111,12 @@ def master_ip():
     return dcos.mesos.DCOSClient().metadata().get('PUBLIC_IPV4')
 
 
+def leader_ip():
```
Comment (alenkacz) [inline shakedown/dcos/__init__.py]: I think there is already shakedown.master_leader_ip()

**Comment by kensipe** [comment]
@alenkacz good catch!  thanks!

**Comment by kensipe** [comment]
@alenkacz could I get another quick review?   It is ready to land with tests and docs

**Comment by kensipe** [comment]
@sschneid what's with all the test failures?   they all seem unrelated.

**Comment by sschneid** [comment]
Failures unrelated to this PR, I've verified manually.

**Pull request merged**